File: matiasmasso/Mat.NET/DTO/DTOLiniaTelefon.cs
using System;

namespace DTO
{
    public class DTOLiniaTelefon : DTOBaseGuid
    {
        public string num { get; set; }
        public string alias { get; set; }
        public DTOContact contact { get; set; }
        public string icc { get; set; }
        public string imei { get; set; }
        public string puk { get; set; }
        public DateTime alta { get; set; }
        public DateTime baixa { get; set; }
        public bool privat { get; set; }


        public DTOLiniaTelefon() : base()
        {
        }

        public DTOLiniaTelefon(Guid oGuid) : base(oGuid)
        {
        }


        public class Consum : DTOBaseGuid
        {
            public DTOLiniaTelefon linia { get; set; }
            public DTODocFile docFile { get; set; }
            public DTOYearMonth yearMonth { get; set; }

            public Consum() : base()
            {
            }

            public Consum(Guid oGuid) : base(oGuid)
            {
            }

            public static DTOLiniaTelefon.Consum Factory(DTOLiniaTelefon oLiniaTelefon, DTOYearMonth oYearMonth, DTODocFile oDocFile)
            {
                DTOLiniaTelefon.Consum retval = new DTOLiniaTelefon.Consum();
                {
                    var withBlock = retval;
                    withBlock.linia = oLiniaTelefon;
                    withBlock.yearMonth = oYearMonth;
                    withBlock.docFile = oDocFile;
                }
                return retval;
            }
        }
    }
}

File: matiasmasso/Mat.NET/DT2/DTOLiniaTelefon.vb
Public Class DTOLiniaTelefon
    Inherits DTOBaseGuid

    Property Num As String
    Property [Alias] As String
    Property Contact As DTOContact
    Property Icc As String
    Property Imei As String
    Property Puk As String
    Property Alta As Date
    Property Baixa As Date
    Property Privat As Boolean


    Public Sub New()
        MyBase.New()
    End Sub

    Public Sub New(oGuid As Guid)
        MyBase.New(oGuid)
    End Sub


    Public Class Consum
        Inherits DTOBaseGuid

        Property Linia As DTOLiniaTelefon
        Property DocFile As DTODocFile
        Property YearMonth As DTOYearMonth

        Public Sub New()
            MyBase.New()
        End Sub

        Public Sub New(oGuid As Guid)
            MyBase.New(oGuid)
        End Sub

        Shared Function Factory(oLiniaTelefon As DTOLiniaTelefon, oYearMonth As DTOYearMonth, oDocFile As DTODocFile) As DTOLiniaTelefon.Consum
            Dim retval As New DTOLiniaTelefon.Consum
            With retval
                .Linia = oLiniaTelefon
                .YearMonth = oYearMonth
                .DocFile = oDocFile
            End With
            Return retval
        End Function
    End Class
End Class


